tools/export/cmake/CMakeLists.txt.tmpl

```diff
@@ -0,0 +1,106 @@
+
+SET(LD_SYS_LIBS "{%- block sys_libs -%} -Wl,--start-group {{ld_sys_libs|join(" ")}} {{libraries|join(" ")}} -Wl,--end-group {%- endblock -%}")
+
+SET(CMAKE_C_FLAGS "{{cc_flags}} -include mbed_config.h")
+SET(CMAKE_CXX_FLAGS "{{cxx_flags}}  -include mbed_config.h")
+SET(CMAKE_ASM_FLAGS "{{asm_flags}}  -include mbed_config.h")
+SET(CMAKE_CXX_LINK_FLAGS "{{ld_flags}} {% for p in library_paths %} {{user_library_flag}}${CMAKE_CURRENT_SOURCE_DIR}/{{p}} {% endfor %}")
+{% if pp -%}
+SET(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} ${LD_SYS_LIBS} {{link_script_option}} ${CMAKE_BINARY_DIR}/{{name}}_pp.link_script.ld")
+{%- endif %}
+
+ADD_DEFINITIONS(
+  {% for d in symbols %}-D{{d}}
+  {% endfor %})
+INCLUDE_DIRECTORIES(
+  {% for p in include_paths %}{{p}}
+  {% endfor %})
+
+# executable {{name}}
+ADD_EXECUTABLE({{name}}
+  {% for src in sources %}{{src}}
+  {% endfor %})
+SET_TARGET_PROPERTIES({{name}} PROPERTIES ENABLE_EXPORTS 1)
+# add syslibs dependencies to create the correct linker order
+TARGET_LINK_LIBRARIES({{name}} {{ld_sys_libs|join(" ")}})
+
+{% if pp -%}
+add_custom_command(TARGET {{name}} PRE_LINK
+                   COMMAND "{{pp}}" {{pp_flags}} {{linker_script}} -o ${CMAKE_CURRENT_BINARY_DIR}/{{name}}_pp.link_script.ld
+                   WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+                   BYPRODUCTS "${CMAKE_CURRENT_BINARY_DIR}/{{name}}_pp.link_script.ld"
+                   )
+{%- endif %}
+
+add_custom_command(TARGET {{name}} POST_BUILD
+                   COMMAND ${ELF2BIN} -O ihex $<TARGET_FILE:{{name}}> $<TARGET_FILE:{{name}}>.hex
+                   COMMAND ${CMAKE_COMMAND} -E echo "-- built: $<TARGET_FILE:{{name}}>.hex"
+                  )
+
+
+##########################################################################
+# mbed-cli specific targets
+##########################################################################
+
+# detect the build type and select the corresponding cli profile
+SET(MBED_BUILD_PROFILE "")
+STRING(TOLOWER ${CMAKE_BUILD_TYPE} LOWERCASE_CMAKE_BUILD_TYPE)
+IF(LOWERCASE_CMAKE_BUILD_TYPE MATCHES debug)
+    SET(MBED_BUILD_PROFILE "mbed-os/tools/profiles/debug.json")
+ELSEIF(LOWERCASE_CMAKE_BUILD_TYPE MATCHES relwithdebinfo)
+    SET(MBED_BUILD_PROFILE "mbed-os/tools/profiles/develop.json")
+ELSEIF(LOWERCASE_CMAKE_BUILD_TYPE MATCHES release)
+    SET(MBED_BUILD_PROFILE "mbed-os/tools/profiles/release.json")
+ELSEIF(LOWERCASE_CMAKE_BUILD_TYPE MATCHES minsizerel)
+    SET(MBED_BUILD_PROFILE "mbed-os/tools/profiles/release.json")
+ELSE()
+  MESSAGE(WARNING "Build type '${CMAKE_BUILD_TYPE}' is unknown, using debug profile")
+  SET(MBED_BUILD_PROFILE "mbed-os/tools/profiles/debug.json")
+ENDIF()
+
+# optional custom target to build via mbed-cli
+ADD_CUSTOM_TARGET(mbed-cli-build
+        COMMAND ${CMAKE_COMMAND} -E echo "mbed compile --build BUILD/${CMAKE_BUILD_TYPE} --profile ${MBED_BUILD_PROFILE}"
+        COMMAND mbed compile --build BUILD/${CMAKE_BUILD_TYPE} --profile ${MBED_BUILD_PROFILE}
```

Comment:
Is this the primary compiling method?

Reply:
No, that’s just for convenience. The primary method is to choose the {{name}} target.

Reply:
Thanks.